### PR TITLE
Add events to upgrade & blocking callbacks

### PR DESCRIPTION
**Pull request by jakearchibald**

### CODE REVIEW

**Review by iflan — APPROVED** [review]
I'm giving the LGTM because I think the changes here are fine, but the `blocked` callback could be improved as well.

File: src/entry.ts [hunk 1]
```diff
@@ -72,7 +83,11 @@ export function openDB<DBTypes extends DBSchema | unknown = unknown>(
   openPromise
     .then((db) => {
       if (terminated) db.addEventListener('close', () => terminated());
-      if (blocking) db.addEventListener('versionchange', () => blocking());
+      if (blocking) {
+        db.addEventListener('versionchange', (event) =>
+          blocking(event.oldVersion, event.newVersion, event),
```
[the remark this answers is not in the document]
Comment (jakearchibald) [reply src/entry.ts:88]: Good catch! Typescript types it as a simple event, so I missed it.

[the remark this answers is not in the document]
Comment (jakearchibald) [reply src/entry.ts:88]: Filed https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1405

**Comment by jakearchibald** [comment]
Published as 7.1.0!

**Pull request merged**